ethcore/src/blockchain/blockchain.rs

```diff
@@ -409,6 +409,27 @@ impl<'a> Iterator for AncestryIter<'a> {
 	}
 }
 
+impl<'a> AncestryIter<'a> {
+	pub fn to_header_iter(self) -> Box<Iterator<Item=ExtendedHeader> + 'a> {
```

Comment:
can and should use `impl Iterator` for this

Reply:
impl trait is not yet stabilized? https://github.com/rust-lang/rust/issues/34511

Reply:
I'll get it another struct implementing `Iterator`. I guess that's clearer.

Reply:
my mistake, thought it was stabilized